cmd/cluster-node-tuning-operator/main.go

```diff
@@ -2,63 +2,133 @@ package main
 	"github.com/openshift/cluster-node-tuning-operator/pkg/signals"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/tuned"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/util"
 	"github.com/openshift/cluster-node-tuning-operator/version"
 )
 
+const (
+	operandFilename  = "openshift-tuned"
+	operatorFilename = "cluster-node-tuning-operator"
+	metricsHost      = "0.0.0.0"
+)
+
 var (
-	boolVersion = flag.Bool("version", false, "show program version and exit")
-	boolLocal   = flag.Bool("local", false, "local run outside a pod")
+	scheme = apiruntime.NewScheme()
 )
 
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(tunedv1.AddToScheme(scheme))
+	utilruntime.Must(mcov1.AddToScheme(scheme))
+	utilruntime.Must(apiconfigv1.Install(scheme))
+}
+
 func printVersion() {
 	klog.Infof("Go Version: %s", runtime.Version())
 	klog.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
 	klog.Infof("%s Version: %s", tunedv1.TunedClusterOperatorResourceName, version.Version)
 }
 
 func main() {
-	const (
-		operandFilename  string = "openshift-tuned"
-		operatorFilename string = "cluster-node-tuning-operator"
-	)
+	var boolVersion bool
+	var enableLeaderElection bool
+	flag.BoolVar(&boolVersion, "version", false, "show program version and exit")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
 
 	runAs := filepath.Base(os.Args[0])
 
-	stopCh := signals.SetupSignalHandler()
-
 	switch runAs {
 	case operatorFilename:
+		var metricsAddr string
+		flag.StringVar(&metricsAddr, "metrics-addr", fmt.Sprintf("%s:%d", metricsHost, metrics.MetricsPort),
+			"The address the metric endpoint binds to.")
+
 		klog.InitFlags(nil)
 		flag.Parse()
 
 		printVersion()
 
-		if *boolVersion {
+		if boolVersion {
 			os.Exit(0)
 		}
 
-		go metrics.RunServer(metrics.MetricsPort, stopCh)
-		metrics.RegisterVersion(version.Version)
-		controller, err := operator.NewController()
+		// We have two namespaces that we need to watch:
+		// 1. NTO namespace - for NTO resources
+		// 2. None namespace - for cluster wide resources
+		ntoNamespace := config.OperatorNamespace()
+		namespaces := []string{
+			ntoNamespace,
+			metav1.NamespaceNone,
+		}
+
+		restConfig := ctrl.GetConfigOrDie()
+		le := util.GetLeaderElectionConfig(restConfig, enableLeaderElection)
+		mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+			NewCache:                cache.MultiNamespacedCacheBuilder(namespaces),
+			Scheme:                  scheme,
+			LeaderElection:          true,
+			LeaderElectionID:        config.OperatorLockName,
+			LeaderElectionNamespace: ntoNamespace,
+			LeaseDuration:           &le.LeaseDuration.Duration,
+			RetryPeriod:             &le.RetryPeriod.Duration,
+			RenewDeadline:           &le.RenewDeadline.Duration,
+			Namespace:               ntoNamespace,
+		})
```

Comment:
By default SDK/Kubebuilder scaffold the projects as cluster-scope, see: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/main.go#L68-L79

But you can change the. scope: https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/